router/http_test.go

```diff
@@ -339,6 +339,34 @@ func (s *S) TestHTTPHeadersFromClient(c *C) {
 	c.Assert(res.StatusCode, Equals, 200)
 }
 
+func (s *S) TestConnectionCloseHeaderFromClient(c *C) {
+	h := http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		// Connection: close header should be stripped by the reverse proxy so it
+		// always does keep-alive with backends.
+		c.Assert(req.Close, Equals, false)
+		c.Assert(req.Header.Get("Proxy-Authenticate"), Equals, "")
+		c.Assert(req.Header.Get("Proxy-Authorization"), Equals, "")
```

Comment:
It looks like "hop-by-hop" got redefined in [RFC 7230](https://tools.ietf.org/html/rfc7230#section-6.1):

> When a header field aside from Connection is used to supply control information for or about the current connection, the sender MUST list the corresponding field-name within the Connection header field.  A proxy or gateway MUST parse a received Connection header field before a message is forwarded and, for each connection-option in this field, remove any header field(s) from the message with the same name as the connection-option, and then remove the Connection header field itself (or replace it with the intermediary's own connection options for the forwarded message).
>
> Hence, the Connection header field provides a declarative way of distinguishing header fields that are only intended for the immediate recipient ("hop-by-hop") from those fields that are intended for all recipients on the chain ("end-to-end"), enabling the message to be self-descriptive and allowing future connection-specific extensions to be deployed without fear that they will be blindly forwarded by older intermediaries.